packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -769,7 +769,15 @@ class _DatePickerHeader extends StatelessWidget {
       maxLines: orientation == Orientation.portrait ? 1 : 2,
       overflow: TextOverflow.ellipsis,
     );
-
+    final Widget? effectiveEntryModeButton;
+    if (entryModeButton != null) {
```

Comment:
This is previous not needed because the `entryModeButton` conflicted with TextField and marked as explicit. After the patch the TextField creates an empty SemanticsConfig with a bunch of explicit child semantics node. so the semantics config no longer conflict and the icon button gets merge to parent and create weird semantics.

This fix makes sure it won't ever merge up